crates/dpapi-transport/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+#![doc = include_str!("../README.md")]
+#![warn(missing_docs)]
+
+mod connect_options;
+
+use std::future::Future;
+use std::io::Result;
+
+pub use connect_options::{
+    ConnectOptions, DEFAULT_RPC_PORT, Error as ConnectOptionsError, GetSessionTokenFn, ProxyOptions,
+};
+
+/// Represents a transport for communicating with the target server.
+pub trait Transport {
+    /// A type that represents communication stream.
+    type Stream: LocalStream;
+
+    /// Connects to the target server.
+    fn connect(connect_options: &ConnectOptions) -> impl Future<Output = Result<Self::Stream>>;
+}
+
+/// Stream for reading and writing bytes.
+pub trait LocalStream {
```

Comment:
question: What is `Local` for?

Reply:
🤔 I don't remember why I named it `Local`. I think I can rename it to just `Stream`

Reply:
> I think I can rename it to just `Stream`

done